ob-sql-mode.el

```diff
@@ -211,26 +211,18 @@ parameters to the code block.")
   "Return the comint buffer for this session.
 
 Determines the buffer from values in PROCESSED-PARAMS."
-  (let* ((bufname (org-babel-sql-mode--buffer-name processed-params))
-         (sql-bufname (format "*SQL: %s*" bufname))
-         (buf (get-buffer sql-bufname))
-         (product (intern (cdr (assoc :product processed-params)))))
-    (unless (assoc product sql-product-alist)
-      (user-error "Product `%s' is not in `sql-product-alist'" product))
-    (save-current-buffer
-      (unless (sql-buffer-live-p buf product)
-        (if (y-or-n-p (format "Interpreter not running in %s.  Start it? "
-                              sql-bufname))
-            ;; Temporarily redefine pop-to-buffer to do nothing, so
-            ;; that when sql-product-interactive calls it nothing
-            ;; happens.  Otherwise the frame is split to show the
-            ;; interactive buffer, which is not wanted.
-            (let ((old-pop-to-buffer (symbol-function 'pop-to-buffer)))
-              (fset 'pop-to-buffer #'(lambda (&rest r)))
-              (sql-product-interactive product bufname)
-              (fset 'pop-to-buffer old-pop-to-buffer))
-          (user-error "Can't do anything without an SQL interactive buffer")))
-      (get-buffer sql-bufname))))
+  (let* ((sql-product (org-babel-sql-mode-as-symbol (cdr (assoc :product processed-params))))
+         (default-buffer (sql-find-sqli-buffer)))
```

Comment:
Thanks for the proposal (and the associated cleanups in this change).

With respect to using the default SQL buffer if it's available, I'm not yet convinced this is a good idea.

My concern is that it would make the results less repeatable -- by using the default buffer you're at the mercy of whatever other commands might already have been run in there.

For example, with SQLite, you can change the default case-insensitive behaviour of the LIKE operator by executing

```
PRAGMA case_sensitive_like = true;
```

If you've executed that in the default buffer at some point in the past, all the queries in that buffer (while it's active) will use it. 

So you might inadvertently write a query using ob-sql-mode that relies on this PRAGMA, but forget to put in a block in the same `:session` in the Org file.

There are other pragmas in SQLite that have similar query-visible effects like `encoding`, `full_column_names`, and `query_only`. I imagine there are similar features in other backends.

I have a similar concern in re prompting the user for the `sql-product` to use. I think either a default that the user sets, or something that's made explicit in the `.org` file with a property is the safest way to protect against inadvertent mistakes.